apps/research-service/src/validations/validation.service.ts

```diff
@@ -35,4 +36,76 @@ export class ValidationService {
       criteriaList
     });
   }
+
+  async getSiteValidations(
+    siteUuid: string,
+    pageSize = MAX_PAGE_SIZE,
+    pageNumber = 1
+  ): Promise<{
+    validations: ValidationDto[];
+    total: number;
+  }> {
+    if (pageSize > MAX_PAGE_SIZE || pageSize < 1) {
+      throw new BadRequestException(`Invalid page size: ${pageSize}`);
+    }
+    if (pageNumber < 1) {
+      throw new BadRequestException(`Invalid page number: ${pageNumber}`);
+    }
+
+    const sitePolygons = await SitePolygon.findAndCountAll({
+      where: {
+        siteUuid,
+        isActive: true
+      },
+      attributes: ["polygonUuid"],
+      limit: pageSize,
+      offset: (pageNumber - 1) * pageSize
+    });
+
+    if (sitePolygons.count === 0) {
+      throw new NotFoundException(`Site with UUID ${siteUuid} not found or has no polygons`);
+    }
```

Comment:
I don't think a 404 here makes sense. For an index endpoint, I usually prefer to simply return 0 results instead. That's a totally valid result for an index and is a little simpler to handle on the FE.